tensorflow/core/kernels/generate_box_proposals_op.cu.cc

```diff
@@ -38,56 +38,66 @@ limitations under the License.
 
 namespace tensorflow {
 typedef Eigen::GpuDevice GPUDevice;
-#define TF_OP_REQUIRES_CUDA_SUCCESS_ASYNC(context, result, done)       \
+#define TF_OP_REQUIRES_CUDA_SUCCESS(context, result)                   \
   do {                                                                 \
     cudaError_t error(result);                                         \
     if (!SE_PREDICT_TRUE(error == cudaSuccess)) {                      \
       context->SetStatus(errors::Internal("Cuda call failed with",     \
                                           cudaGetErrorString(error))); \
-      done();                                                          \
       return;                                                          \
     }                                                                  \
   } while (0)
 
 namespace {
+
+template <bool T>
+__device__ float legacy_op(float);
+template <>
+__device__ float legacy_op<true>(float a) {
+  return a + 1.;
+}
+template <>
+__device__ float legacy_op<false>(float a) {
+  return a;
+}
 // Decode d_bbox_deltas with respect to anchors into absolute coordinates,
 // clipping if necessary.
-// H is height, W is width, A is the number of anchors.
 // prenms_nboxes maximum number of boxes per image to decode.
 // d_boxes_keep_flags mask for boxes to consider in NMS.
 // min_size is the lower bound of the shortest edge for the boxes to consider.
 // bbox_xform_clip is the upper bound of encoded width and height.
-// correct_transform is a flag to apply coordinate transformation correction to
-// x2 and y2.
+template <bool T>
 __global__ void GeneratePreNMSUprightBoxesKernel(
     const Cuda2DLaunchConfig config, const int* d_sorted_scores_keys,
-    const float4* d_bbox_deltas, const float4* d_anchors, const int H,
-    const int W, const int A, const float min_size, const float* d_img_info_vec,
-    const float bbox_xform_clip, const bool correct_transform,
+    const float4* d_bbox_deltas, const float4* d_anchors, const int height,
+    const int width, const int num_anchors, const float min_size,
+    const float* d_img_info_vec, const float bbox_xform_clip,
```

Comment:
Please comment the shape of d_img_info_vec and d_scores.

Reply:
They are described in op documentation but I added it here as well.